enzy_htp/_interface/pymol_interface.py

```diff
@@ -8,77 +8,193 @@
+    
     Atributes:
-        config_ : the PyMolConfig() class which provides settings of PyMol 
+        config_ : The PyMOLConfig() class which provides settings for both running PyMOL and maintaining a compatible environment.
+        env_manager_ : The EnvironmentManager() class which ensures all required environment elements exist.
+        compatible_env_ : A bool() indicating if the current environment is compatible with the object itself.
     """
-
-    def __init__(self, config: PyMolConfig = None) -> None:
-        """Simplistic constructor that optionally takes an PyMolConfig object as its only
-        argument.
+    def __init__(self, parent, config: PyMolConfig = None) -> None:
+        """Simplistic constructor that optionally takes an PyMOLConfig object as its only argument.
+        Calls parent class.
         """
-        self.config_ = config
-        if not self.config_:
-            self.config_ = default_pymol_config()
+        super().__init__(parent, config, default_pymol_config)
+        self.session_ = self.new_session() 
+        self.available_cmds_:List[str] = self.session_.cmd.kw_list
 
-    def new_pymol_session(self) -> pymol2.PyMOL:
-        """create a new pymol session, start it, and set feedback level
-        TODO(qz): is making a pymol_session class a good idea?"""
+    def new_session(self) -> pymol2.PyMOL:
+        """create a new pymol session, start it, and set feedback level. Canonical means to get a session in enzy_htp."""
         result = pymol2.PyMOL()
         result.start()
         result.cmd.feedback(*self.config.DEFAULT_OUTPUT_LV)
         return result
 
+
+    def check_pymol2_installed(self) -> None:
+        """Method that checks if """
+        if "pymol2" in self.missing_py_modules():
+            _LOGGER.errorf("pymol2 is NOT installed. Use 'conda install -c conda-forge -y -q pymol-open-source'. Exiting...")
+            exit( 1 )
+
+
+    def convert(self, file_1:str, file_2:str=None, new_ext:str=None, split_states:bool=False, session:pymol2.PyMOL=None) -> Union[str, List[str]]:
+        """Method that converts a supplied file to a different format. Either a new filename or new file 
+        extension can be supplied. If neither or both are supplied, then the function will exist. Note
+        that the function does not check for valid file types and will catch any errors that are thrown
+        if an invalid file_1 or output file combination is supplied. Returns the new outfile.
+
+        Args:
+            file_1 : The name of the original file as a str().
+            file_2 : The name of the output file as a str(). Optional.
+            new_ext : The new extension to use. Optional.
+            session : A pymol2.PyMOL() session to use. By default uses an internal session instance that resets each function call.
+
+        Returns:
+            The name of the new file as a str().
+        """
+        self.check_pymol2_installed()
```

Comment:
why is this check exist for all methods? Could having it in __init__ so it checks when the singletion object spawns?

Reply:
The reason it exists is that for all other `<Package>Interface`, an executable missing from the environment is not a problem until you try to run the function that uses it. So if the element isn't needed and is missing. it does not matter. `PyMolInterface()` is an exception since it relies on a python package. I wanted to preserve this behavior where the `enzy_htp` package can still run without having every element. So this is my solution.

Reply:
Ohh I see, it exit(1), make sense.